cmd/snap/cmd_run.go

```diff
@@ -530,6 +531,54 @@ func migrateXauthority(info *snap.Info) (string, error) {
 	return targetPath, nil
 }
 
+func activateXdgDocumentPortal(info *snap.Info) error {
+	// Don't do anything for snaps that don't plug the desktop interface
+	plugsDesktop := false
+	for _, plug := range info.Plugs {
```

Comment:
Can you add a comment like this please
```
NOTE: This check is imperfect because we don't really know if the interface is connected
or not but this is an acceptable compromise for not having to communicate with snapd
in snap run. In a typical desktop session the document portal can be in use by many
applications, not just by snaps, so this is at most, pre-emptively using some extra memory.
```